snap/validate_test.go

```diff
@@ -111,6 +111,48 @@ func (s *ValidateSuite) TestValidateName(c *C) {
 	}
 }
 
+func (s *ValidateSuite) TestValidateInstanceName(c *C) {
+	validNames := []string{
+		// plain names are also valid instance names
+		"a", "aa", "aaa", "aaaa",
+		"a-a", "aa-a", "a-aa", "a-b-c",
+		// snap instance
+		"foo_bar",
+		"foo_0000000001",
+		"01game_0000000001",
```

Comment:
I think I'd go with `0123456789` instead of `0000000001` (as you do below in invalid for the maxlength + 1 case) because that way it's easier for me, reading it, to tell you're checking for max length